docs/Hosting_Application_Usage.md

```diff
@@ -0,0 +1,134 @@
+# How to Use ONNX Hosting Application REST API for Prediction
+
+The ONNX hosting application is designed to provide a REST API for a prediction service. The command line is quite simple as below. Only `model_path` is MUST-HAVE to start a server.
+
+```
+~./onnxruntime_hosting --help
+ONNX Hosting: host an ONNX model for inferencing with ONNXRuntime
+Allowed options:
+  -h [ --help ]                   Shows a help message and exits
+  --logging_level arg (=verbose)  Logging level. Allowed options (case 
+                                  sensitive): verbose, info, warning, error, 
+                                  fatal
+  -m [ --model_path ] arg         Path to ONNX model
+  -a [ --address ] arg (=0.0.0.0) The base HTTP address
+  --http_port arg (=8001)         HTTP port to listen to requests
+  --num_http_threads arg (=8)     Number of http threads
+
+```
+
+## Start the Server
+
+To host an ONNX model as REST API server, you simply need to run as:
+
+```
+./onnxruntime_hosting -m /<your>/<model>/<path>
+```
+
+The prediction URL is in this format:
+
+```
+http://<your_ip_address>:<port>/v1/models/<your-model-name>/versions/<your-version>:predict
+```
+
+**Note**: Since we currently only support one model, the model name and version can be any string length > 0.
+
+## Request and Response Payload
+
+The HTTP request can be a protobuf message in binary format or JSON format. The HTTP request header field `Content-Type` will tell the server how to handle the request payload. It is a MUST-HAVE field in request headers. Requests missing `Content-Type` will be rejected as `400 Bad Request`.
```

Comment:
IS this be the correct behavior?

Reply:
This is expected. Otherwise, we could not know how to deserialize the payload. Some of Azure RPs and AzureML MMS having the same behavior.